### PR TITLE
MISUV 5499 provide default data for api 1885

### PR DESCRIPTION
**Pull request by simondrugan16**

### CODE REVIEW

File: app/controllers/CalculationController.scala [hunk 1]
```diff
@@ -35,12 +34,35 @@ import scala.concurrent.Future
...
+    val id = s"/income-tax/view/calculations/liability/$nino/$calcId"
+    dataRepository
+      .find(equal("_id", id), equal("method", GET))
+      .flatMap {
+        case stubData@Some(dataModel: DataModel) =>
+          dataModel.response match {
+            case Some(_: JsValue) => Future(Status(stubData.head.status)(stubData.head.response.get))
+            case None => logger.info(s"[CalculationController][getCalcLegacy] " +
+              s"Could not find endpoint in Dynamic Stub matching the URI: $id . Calling fallback default endpoint.")
+              Future {
```
Comment (joesheridan) [inline app/controllers/CalculationController.scala:52]: FYI normally we'd use Future.successful(..) here

File: app/controllers/CalculationController.scala [hunk 2]
```diff
@@ -35,12 +34,35 @@ import scala.concurrent.Future
...
+                Status(NO_CONTENT)
+              }
+          }
+        case None =>
+          logger.info(s"[CalculationController][getCalcLegacy] " +
+            s"Could not find endpoint in Dynamic Stub matching the URI: $id . Calling fallback default endpoint.")
+          val fallbackUrl: String = "/income-tax/view/calculations/liability/SUCCESS1A/041f7e4d-87d9-4d4a-a296-3cfbdf2023m6"
```
[the remark this answers is not in the document]
Comment (simondrugan16) [reply app/controllers/CalculationController.scala:59]: I added a file with this URI (_id) in ATs and it worked fine. Waiting for Himani to make a PR with test data then will make sure the one in her PR matches this one. Was considering putting the URL in a utils fine or something too as it's not particularly readable like this

File: app/repositories/DefaultValues.scala [hunk 3]
```diff
@@ -49,4 +56,15 @@ object DefaultValues extends Logging {
...
   }
+
+  def getDefaultRequestHandler(url: String): Future[Result] = {
+    dataRepository
+      .find(equal("_id", url), equal("method", GET))
+      .map {
+        case stubData@Some(_: DataModel) =>
+          Status(stubData.head.status)(stubData.head.response.get)
+        case None =>
+          NotFound(s"Failed to find the default API endpoint in the repository matching the URI: $url")
```
Comment (avelx) [inline app/repositories/DefaultValues.scala:67]: You can extract required values in a safe way:

`      .map {
        case Some(DataModel(_, _, _, status, Some(response))) =>
          Status(status)(response)
        case None =>
          NotFound(s"Failed to find the default API endpoint in the repository matching the URI: $url")
      }`

**Review by avelx — APPROVED** [review]
approved

**Pull request merged**